packages/react-server/core/renderMiddleware.js

```diff
@@ -590,18 +592,27 @@ function renderScripts(pageObject, res) {
 }
 
 function renderStylesheets (pageObject, res) {
-	pageObject.getHeadStylesheets().forEach((styleSheet) => {
-		if (styleSheet.href) {
-			res.write(`<link rel="stylesheet" type="${styleSheet.type}" media="${styleSheet.media}" href="${styleSheet.href}" ${PAGE_CSS_NODE_ID}>`);
-		} else if (styleSheet.text) {
-			res.write(`<style type="${styleSheet.type}" media="${styleSheet.media}" ${PAGE_CSS_NODE_ID}>${styleSheet.text}</style>`);
-		} else {
-			throw new Error("Style cannot be rendered because it has neither an href nor a text attribute: " + styleSheet);
-		}
+
+	const styles = PageUtil.standardizeStyles(pageObject.getHeadStylesheets());
+
+	const stylePromises = styles.map(styleP => {
```

Comment:
This needs to be a _chain_, right?  `<link/>` tags could get written out of order with this map.